lib/awful/spawn.lua

```diff
@@ -79,6 +79,52 @@
 --        placement = awful.placement.bottom_right,
 --    })'
 --
+-- **Getting a command result**:
+--
+-- First, do **not** use `io.popen` **ever**. It is synchronous. Synchronous
+-- function **stop everything** until they are done. All visual applications
+-- locks (as Awesome no longer respond), you will probably lose some keyboard
+-- and mouse events and will have higher latency when playing games. This is
```

Comment:
> (but some key-bindings and focus-changing-keybindings will have non-intuitive results).

Or not happen at at. I have this issue when I compile and move the mouse too fast (yes, some modules I use, like Vicious and Drawer, still use `popen`...). This is the kind of issues I am referring to. I found awesome to be unreliable when it is slowed down by blocking calls. My Awesome 3.4 config had serious issues with that.